### PR TITLE
FAI-1227 - Store LB access logs in s3

### PR DESCRIPTION
**Pull request by alexisschwartz**
## Description

Stores LB access logs in S3

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

https://faros-ai.atlassian.net/browse/FAI-1227

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -595,3 +595,14 @@ variable "admin_user" {
...
+
+variable "lb_logging_bucket" {
+  description = "The s3 bucket which LB access logs should be stored to"
+  type        = string
+}
+
+variable "external_lb_logging_prefix" {
+  description = "s3 prefix for the external LB access logs"
+  type        = string
+  default     = "kong-external-lb"
```
Comment (tovbinm) [inline variables.tf]: remove the default

File: lb.tf [hunk 2]
```diff
@@ -52,6 +52,11 @@ resource "aws_lb" "external" {
     create = var.lb_creation_timeout
     delete = var.lb_deletion_timeout
   }
+  access_logs {
```
Comment (tovbinm) [inline lb.tf]: lets use a dynamic block here - https://www.terraform.io/language/expressions/dynamic-blocks

example - https://github.com/faros-ai/terraform-aws-faros/blob/c543ccb45eb36a817c1b14ba5d8a8c761765ca2c/vpc.tf#L72

e.g
```terraform
dynamic "access_logs" {  

  for_each = toset(length(var.lb_logging_bucket) > 0 ? [var.lb_logging_bucket] : [])
  content {
     bucket = access_logs.value
     prefix = length(var.external_lb_logging_prefix) > 0 ? var.external_lb_logging_prefix : format("%s-%s-external", var.service, var.environment)
     enabled = length(var.lb_logging_bucket) > 0
  }
}

```

File: lb.tf [hunk 3]
```diff
@@ -52,6 +52,14 @@ resource "aws_lb" "external" {
     create = var.lb_creation_timeout
     delete = var.lb_deletion_timeout
   }
+  dynamic "access_logs" {
+    for_each = toset(length(var.lb_logging_bucket) > 0 ? [lb_logging_bucket] : [])
```
Comment (tovbinm) [inline lb.tf]: `[var.lb_logging_bucket]`

File: lb.tf [hunk 4]
```diff
@@ -52,6 +52,14 @@ resource "aws_lb" "external" {
...
+    content {
+      bucket = access_logs.value
+      prefix = length(var.external_lb_logging_prefix) > 0 ? var.external_lb_logging_prefix : format("%s-%s-external", var.service, var.environment)
+      enabled = length(var.lb_logging_bucket) > 0
```
Comment (tovbinm) [inline lb.tf]: this can be `enabled = true`

File: variables.tf [hunk 5]
```diff
@@ -595,3 +595,14 @@ variable "admin_user" {
   type        = string
   default     = "kong-admin"
 }
+
+variable "lb_logging_bucket" {
+  description = "The s3 bucket which LB access logs should be stored to"
+  type        = string
+}
+
+variable "external_lb_logging_prefix" {
```
Comment (tovbinm) [inline variables.tf:604]: Remove the default value for this var

**Review by tovbinm — APPROVED** [review]
Just one comment left, the Therese lgtm

File: variables.tf [hunk 6]
```diff
@@ -595,3 +595,13 @@ variable "admin_user" {
   type        = string
   default     = "kong-admin"
 }
+
+variable "lb_logging_bucket" {
```
Comment (tovbinm) [inline variables.tf:599]: Oh my mistake! we need to set the defaults to `lb_logging_bucket` and `external_lb_logging_prefix` to `""`

Comment (alexisschwartz) [reply variables.tf:599]: Good catch. Thanks. PR incoming!

Comment (alexisschwartz) [reply variables.tf:599]: https://github.com/faros-ai/terraform-aws-kong/pull/10

**Pull request merged**